src/components/add.js

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useEffect,useContext } from 'react';
+import { Text, View, TextInput, StyleSheet, FlatList, TouchableOpacity, Image,Button } from 'react-native';
+import { GlobalContext } from "../context/GlobalState";
+
+const Add = () => {
+  const [filterdData, setFilterdData] = useState([]);
+  const [results, setResults] = useState([]);
+  const [query, setQuery] = useState("");
+  const {
+    addMovieToFavouritelist,
+  } = useContext(GlobalContext);
+
+  useEffect(() => {
+    fetchMovie();
+    return () => {
+
+    }
+  }, [])
+  const fetchMovie = () => {
+  
+    fetch(
+      `https://api.themoviedb.org/3/movie/popular?api_key=29d62274e14964e04e0a98c5c473c7b1&language=en-US&page=1`
+    )
+      .then((res) => res.json())
+      .then((data) => {
+        setFilterdData(data.results);
+        setResults(data.results);
+      }).catch((error) => {
+        console.error(error);
+      })
+  }
+
+  const onChange = (e) => {
+    e.preventDefault();
+
+    setQuery(e.target.value);
+
+    fetch(
+      `https://api.themoviedb.org/3/search/movie?api_key=29d62274e14964e04e0a98c5c473c7b1&language=en-US&page=1&include_adult=false&query=${e.target.value}`
```

Comment:
Ideally, we store the main website URL in an environment variable/constant and then use it where needed and append the remaining URL as required